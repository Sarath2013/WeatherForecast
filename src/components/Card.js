import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';


const CardComponent = (props) => {

    function setActiveChartData() {
        props.onSelect(props.cardData);
    }

    return (
        <Card className="card-comp" onClick={setActiveChartData}>
            <CardActionArea>
                <CardContent>
                    <Typography className="card-name" gutterBottom>
                        Temp:
                    </Typography>
                    <Typography className="card-value" gutterBottom>
                        {(props.temperatureType === 'celsius') ? props.cardData.tempC : props.cardData.tempF}
                    </Typography>
                    <Typography className="card-name" gutterBottom>
                        Date:
                    </Typography>
                    <Typography className="card-value" gutterBottom>
                        {props.cardData.displayDate}
                    </Typography>
                    <Typography className="card-name" gutterBottom>
                        Humidity:
                         </Typography>
                    <Typography className="card-value" gutterBottom>
                        {props.cardData.humidity}
                    </Typography>
                    <Typography className="card-name" gutterBottom>
                        Weather:
                    </Typography>
                    <Typography gutterBottom className="content-ellipsis card-value">
                        {props.cardData.weather}
                    </Typography>
                </CardContent>
            </CardActionArea>
        </Card>
    );
}

CardComponent.propTypes = {
    cardData: PropTypes.shape({
        chartDataC: PropTypes.array.isRequired,
        chartDataF: PropTypes.array.isRequired,
        date: PropTypes.string.isRequired,
        displayDate: PropTypes.string.isRequired,
        humidity: PropTypes.number.isRequired,
        tempC: PropTypes.string.isRequired,
        tempF: PropTypes.string.isRequired,
        weather: PropTypes.string.isRequired,
        weatherArr: PropTypes.array.isRequired,
        fragmentsC: PropTypes.array.isRequired,
        fragmentsF: PropTypes.array.isRequired,
    }).isRequired,
    onSelect: PropTypes.func.isRequired,
    temperatureType: PropTypes.string.isRequired,
}

export default React.memo(CardComponent);