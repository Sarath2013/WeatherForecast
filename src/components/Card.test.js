import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Card from './Card';

let wrapper, cardData;

cardData = {
    date: "2019-11-12",
    displayDate: "12 Nov 19",
    humidity: 83,
    tempC: "5C",
    tempF: "40F",
    weather: "overcast clouds"
}

describe("Unit Testing - Rendering Card with Fahrenheit", () => {
    beforeAll(() => {
    });

    afterAll(() => {
        wrapper = null
    });

    test('renders snapshot', () => {
        wrapper = shallow(<Card cardData={cardData} temperatureType="fahrenheit" />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('ensure card details with Fahrenheit', () => {
        expect(wrapper.find('.card-name').get(0).props.children).toEqual('Temp:');
        expect(wrapper.find('.card-name').get(1).props.children).toEqual('Date:');
        expect(wrapper.find('.card-name').get(2).props.children).toEqual('Humidity:');
        expect(wrapper.find('.card-name').get(3).props.children).toEqual('Weather:');
        expect(wrapper.find('.card-value').get(0).props.children).toEqual('40F');
        expect(wrapper.find('.card-value').get(1).props.children).toEqual('12 Nov 19');
        expect(wrapper.find('.card-value').get(2).props.children).toEqual(83);
        expect(wrapper.find('.card-value').get(3).props.children).toEqual('overcast clouds');
        
    });
});

describe("Unit Testing - Rendering Card with Celsius", () => {
    beforeAll(() => {
    });

    afterAll(() => {
        wrapper = null
    });

    test('renders snapshot', () => {
        wrapper = shallow(<Card cardData={cardData} temperatureType="celsius" />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('ensure card details with Celsius', () => {
        expect(wrapper.find('.card-name').get(0).props.children).toEqual('Temp:');
        expect(wrapper.find('.card-name').get(1).props.children).toEqual('Date:');
        expect(wrapper.find('.card-name').get(2).props.children).toEqual('Humidity:');
        expect(wrapper.find('.card-name').get(3).props.children).toEqual('Weather:');
        expect(wrapper.find('.card-value').get(0).props.children).toEqual('5C');
        expect(wrapper.find('.card-value').get(1).props.children).toEqual('12 Nov 19');
        expect(wrapper.find('.card-value').get(2).props.children).toEqual(83);
        expect(wrapper.find('.card-value').get(3).props.children).toEqual('overcast clouds');
        
    });
});