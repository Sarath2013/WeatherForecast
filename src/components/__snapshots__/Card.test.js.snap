// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing - Rendering Card with Celsius renders snapshot 1`] = `
<WithStyles(ForwardRef(Card))
  className="card-comp"
  onClick={[Function]}
>
  <WithStyles(ForwardRef(CardActionArea))>
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        className="card-name"
        gutterBottom={true}
      >
        Temp:
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="card-value"
        gutterBottom={true}
      >
        5C
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="card-name"
        gutterBottom={true}
      >
        Date:
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="card-value"
        gutterBottom={true}
      >
        12 Nov 19
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="card-name"
        gutterBottom={true}
      >
        Humidity:
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="card-value"
        gutterBottom={true}
      >
        83
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="card-name"
        gutterBottom={true}
      >
        Weather:
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="content-ellipsis card-value"
        gutterBottom={true}
      >
        overcast clouds
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
</WithStyles(ForwardRef(Card))>
`;

exports[`Unit Testing - Rendering Card with Fahrenheit renders snapshot 1`] = `
<WithStyles(ForwardRef(Card))
  className="card-comp"
  onClick={[Function]}
>
  <WithStyles(ForwardRef(CardActionArea))>
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        className="card-name"
        gutterBottom={true}
      >
        Temp:
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="card-value"
        gutterBottom={true}
      >
        40F
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="card-name"
        gutterBottom={true}
      >
        Date:
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="card-value"
        gutterBottom={true}
      >
        12 Nov 19
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="card-name"
        gutterBottom={true}
      >
        Humidity:
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="card-value"
        gutterBottom={true}
      >
        83
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="card-name"
        gutterBottom={true}
      >
        Weather:
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="content-ellipsis card-value"
        gutterBottom={true}
      >
        overcast clouds
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
</WithStyles(ForwardRef(Card))>
`;
